### PR TITLE
Web3 eth contract types

### PR DESCRIPTION
**Pull request by joshstevens19**
web3-eth-contract types.

### CODE REVIEW

File: packages/web3-eth-contract/package.json [hunk 1]
```diff
@@ -8,10 +8,12 @@
     "main": "dist/web3-eth-contract.cjs.js",
     "module": "dist/web3-eth-contract.esm.js",
     "browser": "dist/web3-eth-contract.umd.js",
+    "types": "types",
     "scripts": {
         "build": "rollup -c",
         "dev": "rollup -c -w",
-        "test": "jest"
+        "test": "dtslint types && jest",
```
Comment (nivida) [inline packages/web3-eth-contract/package.json]: Could you do the same change here as described in the accounts PR?

File: packages/web3-eth-contract/types/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    constructor(
+        provider: AbstractProviderAdapter | provider,
+        jsonInterface: {},
+        address?: string,
+        options?: ContractOptions
+    )
+    options: Options;
+    clone(): Contract;
+    deploy(options: DeployOptions): DeployTransactionResponse;
+    methods: any;
```
Comment (nivida) [inline packages/web3-eth-contract/types/index.d.ts:32]: The correct definition would be: ```methods: Function[]```

File: packages/web3-eth-contract/types/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    once(event: string, callback: (error: Error, event: EventData) => void): void;
+    once(event: string, options: EventOptions, callback: (error: Error, event: EventData) => void): void;
+    events: any;
```
Comment (nivida) [inline packages/web3-eth-contract/types/index.d.ts:35]: The correct definition would be: ```events: Function[]```

**Review by nivida — CHANGES_REQUESTED** [review]

File: packages/web3-eth-contract/types/index.d.ts (hunk 2 above)
Comment (joshstevens19) [reply packages/web3-eth-contract/types/index.d.ts:32]: As discussed offline has to stay `any` due to not knowing what the method is being executed i.e. methods.josh() in theory `josh` in compile time will be `any` not `Function[]` so will not work 👍

File: packages/web3-eth-contract/types/index.d.ts (hunk 3 above)
Comment (joshstevens19) [reply packages/web3-eth-contract/types/index.d.ts:35]: As discussed offline has to stay `any` due to not knowing what the method is being executed i.e. events.josh() in theory `josh` in compile time will be `any` not `Function[]` so will not work 👍

**Comment by joshstevens19** [comment]
fixed the `test` script @nivida 👍

**Pull request merged**